p4-16/spec/P4-16-spec.mdk

```diff
@@ -4946,21 +4946,69 @@ switchCase
 
 switchLabel
     : name
+    | expression
     | DEFAULT
     ;
 ~ End P4Grammar
 
-The expression within the `switch` statement is restricted to be
-the result of a table invocation (See Section [#sec-invoke-mau]).
-
-If a switch label is not followed by a block statement it falls
+The expression within the `switch` statement must be one of:
```

Comment:
I would probably make two separate sub-sections for these, I think that the explanations would be easier to follow.

Reply:
I have made a try at following this suggestion, but I do not know if it is any easier to follow.  Take a look and see what you think.